src/hotspot/os/linux/cgroupV2Subsystem_linux.cpp

```diff
@@ -316,6 +316,10 @@ char* CgroupV2Controller::construct_path(char* mount_path, const char* cgroup_pa
   ss.print_raw(mount_path);
   if (strcmp(cgroup_path, "/") != 0) {
     ss.print_raw(cgroup_path);
+    if (strstr((char*)cgroup_path, "../") != nullptr) {
+      log_warning(os, container)("Cgroup v2 path at [%s] is [%s], cgroup limits can be wrong.",
+        mount_path, cgroup_path);
+    }
```

Comment:
Removed extra (char*) cast.

> We should probably move this warning to `CgroupUtil::adjust_controller`, right before we've determined that we actually need to adjust. I wonder, though, if we should just print the warning and set the cgroup_path to `/` and return early. Otherwise, path adjustment will run with no different result.

"../" only appears in corner case with cgroupns=private and the process moved to the outer group. In that specific case we should avoid concatenating with whatever starts with "../".